nireports/tests/test_reportlets.py

```diff
@@ -343,21 +344,33 @@ def test_mriqc_plot_mosaic(tmp_path, test_data_package, outdir, views, plot_sagi
 
     fname = f"mosaic_{'_'.join(v or 'none' for v in views)}_{plot_sagittal:d}.svg"
 
-    testfunc = partial(
-        plot_mosaic,
-        get("MNI152NLin6Asym", resolution=2, desc="LR", suffix="T1w"),
-        plot_sagittal=plot_sagittal,
-        views=views,
-        out_file=(outdir / fname) if outdir is not None else None,
-        title=f"A mosaic plotting example: views={views}, plot_sagittal={plot_sagittal}",
-        maxrows=5,
-    )
+    def _test_mriqc_plot_mosaic(outdir, views, plot_sagittal):
+        testfunc = partial(
+            plot_mosaic,
+            get("MNI152NLin6Asym", resolution=2, desc="LR", suffix="T1w"),
+            plot_sagittal=plot_sagittal,
+            views=views,
+            out_file=(outdir / fname) if outdir is not None else None,
+            title=f"A mosaic plotting example: views={views}, plot_sagittal={plot_sagittal}",
+            maxrows=5,
+        )
 
-    if views[0] is None or ((views[1] is None) and (views[2] is not None)):
-        with pytest.raises(RuntimeError):
+        if views[0] is None or ((views[1] is None) and (views[2] is not None)):
+            with pytest.raises(RuntimeError):
+                testfunc()
+        else:
             testfunc()
+
+    if plot_sagittal:
+        with warnings.catch_warnings():
```

Comment:
What about:

```python
if views[0] ...
    with pytest.raises(RuntimeError):
        testfunc()
elif plot_sagittal:
    with pytest.warns(UserWarning, r".*plot_sagittal.*should not be used")
        testfunc()
else:
    testfunc()
```

Alternately:

```python
if views[0] ...
    context = pytest.raises(RuntimeError)
elif plot_sagittal:
    context = pytest.warns(UserWarning, r".*plot_sagittal.*should not be used")
else:
    context = contextlib.nullcontext()

with context:
    testfunc()
```

If we took the second approach, we could get rid of the `partial()` and just do:

```python
with context:
    plot_mosaic(
        get("MNI152NLin6Asym", resolution=2, desc="LR", suffix="T1w"),
        ...
    )
```

Reply:
a7c643d. Thanks for the review and suggestion Chris.